src/pydata_sphinx_theme/__init__.py

```diff
@@ -501,6 +502,50 @@ def get_edit_url():
     context["theme_show_toc_level"] = int(context.get("theme_show_toc_level", 1))
 
 
+# -----------------------------------------------------------------------------
+# handle pygment css
+# -----------------------------------------------------------------------------
```

Comment:
Can you provide a comment block that describes what is going on here? Either as comments in this code, or as a section in the contributing documentation?

I *think* what is happening is:

- It's impossible to tell Sphinx "use one pygments theme for light mode, and another pygments theme for dark mode".
- So, at build time, we *overwrite* the `pygments.css` file that Sphinx generates.
- Here, we generate two different set of pygments CSS: one for light more and one for dark mode, prefixed by `[data-theme="light"]` etc
- That way, depending on the `data-theme` variable, pygments will choose one or the other set of CSS styles.

Is that right? If so, it'd be helpful to explain that somewhere.